yogstation/code/game/objects/structures/fireaxe.dm

```diff
@@ -88,22 +88,22 @@
 		locked = !locked
 		update_appearance(UPDATE_ICON)
 
-/obj/structure/fireaxecabinet/emag_act(mob/user)
-	//this allows you to emag the fireaxe cabinet, unlocking it immediately.
+/obj/structure/fireaxecabinet/emag_act(mob/user, obj/item/card/emag/emag_card)
 	if(obj_flags & EMAGGED)
-		return
-	if(!open && locked)
-		user.visible_message(span_warning("Sparks fly out of the [src]'s locking modules!"), \
-							 span_caution("You short out the [name]'s locking modules."), \
-							 span_italics("You hear electricity arcing."))
-		spark_system.start()
-
-		src.add_fingerprint(user)
-		obj_flags |= EMAGGED
-		desc += "<BR>[span_warning("Its access panel is smoking slightly.")]"
-
-		playsound(loc, "sound/machines/locktoggle.ogg", 30, 1, -3)
-		locked = 0
-		audible_message("You hear an audible clunk as the [name]'s bolt retracts.")
-		update_appearance(UPDATE_ICON)
-		//Fireaxe Cabinet is now permanently unlocked.
+		return FALSE
+	if(open || !locked) // You won't need the emag to get what you want. Just open it normally!
+		return FALSE
+	// This allows you to emag the fireaxe cabinet, unlocking it immediately.
+	user.visible_message(span_warning("Sparks fly out of the [src]'s locking modules!"), \
+							span_caution("You short out the [name]'s locking modules."), \
+							span_italics("You hear electricity arcing."))
+	spark_system.start()
+	src.add_fingerprint(user)
+	obj_flags |= EMAGGED
+	desc += "<BR>[span_warning("The access panel is smoking slightly.")]"
```

Comment:
This should be in ``update_desc``, and again remove the arg from update_appearance below